snuba/datasets/discover.py

```diff
@@ -30,18 +31,42 @@
 from snuba.query.timeseries import TimeSeriesExtension
 from snuba.request.request_settings import RequestSettings
 from snuba.util import is_condition
+from snuba.utils.metrics.backends.wrapper import MetricsWrapper
 
 EVENTS = "events"
 TRANSACTIONS = "transactions"
 
+metrics = MetricsWrapper(environment.metrics, "api.discover")
+
 
 def detect_table(
-    query: Query, events_only_columns: ColumnSet, transactions_only_columns: ColumnSet
+    query: Query,
+    events_only_columns: ColumnSet,
+    transactions_only_columns: ColumnSet,
+    track_bad_queries: bool,
 ) -> str:
     """
     Given a query, we attempt to guess whether it is better to fetch data from the
     "events" or "transactions" storage. This is going to be wrong in some cases.
     """
+    event_column = None
+    transaction_column = None
+    all_referenced_columns = query.get_all_referenced_columns()
+    for col in all_referenced_columns:
+        if events_only_columns.get(col):
+            event_column = col
+        elif transactions_only_columns.get(col):
+            transaction_column = col
+
+    if event_column is not None and transaction_column is not None:
+        metrics.increment(
```

Comment:
This will work, but I suspect it may be confusing since, when there are multiple incompatible columns we will only record one per table. When we break down by tag in the dashboard, we would see underestimated results.

What about this:
- log if the query is ok (so we can have rate of impossible queries vs possible). Different key, not via tag
- log impossible query by distinguishing these cases through tags: transactions column when picking event tables, events columns when picking transactions.

This way, breaking by tags will give us consistent results vs if we did not break by tags.
The whole list of columns will be on sentry if you also log a warning.